common/src/main/java/com/microsoft/identity/common/internal/util/FileUtil.java

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.util;
+
+import android.content.Context;
+
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.NonNull;
+
+import static com.microsoft.identity.common.java.exception.ClientException.IO_ERROR;
+
+/**
+ * Utility class for reading/writing file in Android.
+ */
+public class FileUtil {
+    private static final String TAG = FileUtil.class.getSimpleName();
+
+    private FileUtil() {
+    }
+
+    /**
+     * Write a data blob to a file.
+     *
+     * @param data             a data blob to be written.
+     * @param context          a {@link Context} object.
+     * @param relativeFilePath a relative path (under the calling app's directory) to a file to write to.
+     */
+    public static void writeDataToFile(@NonNull final byte[] data,
+                                       @NonNull final Context context,
```

Comment:
You could make this non-Android-specific by having the caller provide the `File` itself, rather than a name and a `Context`